# ES0001: Fix invalid 'suffix' argument

| Property | Value |
| --- | --- |
| Id | ES0001 |
| Category | Naming |
| Severity | Error |

## Cause

Scribe constructor argument `suffix` contains at least one character that's illegal after the first position in a type member identifier.

## Rule description

As Scribe appends `suffix` text to existing identifiers to create new or implement existing type members, the resulting string must also be a valid identifier.

Further information on valid type member identifiers may be found in the latest version of the C# language standard ([ECMA-334](https://www.ecma-international.org/publications-and-standards/standards/ecma-334/))

## How to fix rule violations

Remove illegal characters from the `suffix` constructor argument.

Identifier permutations may be quickly tested by creating a type member with the chosen combined text and attempting to compile, similar to the below:

```csharp
class Foo
{
    int @Bar { get; set; }      // Legal
    int _Bar_Bar_ { get; set; } // Legal

    int A_üêü { get; set; }      // üêü is not a legal identifier character
    int @Bar@Bar { get; set; }  // '@' may only appear in the first position
}
```

## When to suppress this rule

Do not suppress this rule. It's a compile-time error and will prevent the build from succeeding.

## Example

### Code with error

```csharp
[Scribe("@Bazz")] // ES0001
partial class Foo
{
    MyEnum Bar { get; set; }
}
```

### Code with fix

```csharp
[Scribe("Bazz")] // Illegal '@' character removed from suffix
partial class Foo
{
    MyEnum Bar { get; set; }
}
```
