# ES0004: Fix scribed member naming collision

| Property | Value |
| --- | --- |
| Id | ES0004 |
| Category | Naming |
| Severity | Error |

## Cause

Scribe type member identifier already exists and isn't a valid implementable method.

## Rule description

If an existing type member identifier matches the combination of enum identifier and `suffix`, the member must be a partial method returning a contextually appropriate string.

For example, a valid collision:

```csharp
[Scribe("Bazz")]
partial class Foo
{
    MyEnum? Bar { get; set; }
    partial string? BarBazz();
}
```

All other member declarations for "BarBazz" are invalid in this example.

## How to fix rule violations

Adjust `suffix` or colliding type member identifier ensure uniqueness.

## When to suppress this rule

Do not suppress this rule. It's a compile-time error and will prevent the build from succeeding.

## Example

### Code with error

```csharp
[Scribe("Bazz")]
partial class Foo
{
    MyEnum Bar { get; set; } // ES0004
    object BarBazz { get; set; }
}
```

### Code with fix (changed suffix)

```csharp
[Scribe("Fish")] // BarFish identifier is unique
partial class Foo
{
    MyEnum Bar { get; set; }
    object BarBazz { get; set; }
}
```

### Code with fix (changed member name)

```csharp
[Scribe("Bazz")]
partial class Foo
{
    MyEnum Bar { get; set; }
    object Fish { get; set; } // BarBazz identifier freed up
}
```
